compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -767,6 +769,27 @@ object Types {
           (name, buf) => buf ++= nonPrivateMember(name).altsWith(_.is(Deferred)))
     }
 
+    /**
+     * Returns the set of methods that are abstract and do not overlap with any of
+     * [[java.lang.Object]] methods.
+     *
+     * Conceptually, a SAM (functional interface) has exactly one abstract method.
+     * If an interface declares an abstract method overriding one of the public
+     * methods of [[java.lang.Object]], that also does not count toward the interface's
+     * abstract method count.
+     *
+     * @see https://docs.oracle.com/javase/8/docs/api/java/lang/FunctionalInterface.html
+     *
+     * @return the set of methods that are abstract and do not match any of [[java.lang.Object]]
+     *
+     */
+    final def possibleSamMethods(implicit ctx: Context): Seq[SingleDenotation] = {
+      record("possibleSamMethods")
+      abstractTermMembers
+        .filterNot(_.symbol.matchingMember(defn.ObjectType).exists)
```

Comment:
You can  use only one filterNot call with two conditions

Reply:
Sorry, I didn't understand, do I need to use a single `filterNot` call with two conditions here?

Reply:
Yes, that's more efficient since you're only doing one traversal

Reply:
@smarter, I amended the method implementation and squashed all the commits into a single one. The PR is ready to be merged.